Added note about masked content to requester

lucascumsille opened the pull request:
## Relevant issue(s)
Fixes: https://github.com/mysociety/whatdotheyknow-theme/issues/2015

## What does this do?
The message advices the requester that emails and phone numbers are automatically redacted from responses. This should help reducing the amount of emails that the authorities receive from requesters asking why those details have been redacted.

## Why was this needed?

## Implementation notes
@gbp 

- Initially I added this conditional`<% if @is_owning_user %>`, but I wonder if we also should include another argument that only display the message if there is a response.

- Ideally we don't want to display the "We work to defend the right to FOI for everyone" message at the same time as the warning message. I have included a conditional `<% if show_donation_button? && !flash[:request_sent] && !@is_owning_user %>`, but this conditional might change depending on the point above.

## Screenshots
<img width="765" alt="Screenshot 2025-07-03 at 13 46 36" src="https://github.com/user-attachments/assets/0f2a74a9-9f2d-49b6-802e-77f59a4276d8" />

## Notes to reviewer

gbp commented:
Nice one @lucascumsille. I'm not sure about the conditionals but maybe @zarino might have some thoughts.

I think with the current changes the donation block isn't rendered for the requester at all so we need to consider if we want to display these messages at the same time as each other, I personally think we should show the donation block. Also @lucascumsille suggests we could, "include another argument that only display the [new redaction warning] message if there is a response" this seems sensible to me too.

garethrees commented:
> (I wonder whether people might skip over the red banner between messages? eg: Do we send email alerts that automatically jump down to the last message?)

Ah yes, we do and I think that would end up getting missed.

> Love the idea of putting it inside the last message!

Yeah looking at them side by side that's what I'd go for I think.

lucascumsille commented:
- I ended up creating a version that mixes @garethrees and @mdeuk solutions. The message is still inside the incoming message, but now I have moved it to the bottom as @mdeuk suggested. I think it makes sense considering the top side of the message already has a prominent header, so probably the warning message would have been missed.

- One thing I noticed and I changed in this PR is the donation CTA, at first glance I thought it was an error message because of the red colour. I think using green looks more positive.(@garethrees let me know if you like this change and if you prefer me to move it to another PR) 

<img width="1481" height="735" alt="Screenshot 2025-07-30 at 14 17 25" src="https://github.com/user-attachments/assets/b94ffe32-6e4c-4ea9-852e-9b530d0dd0c1" />

<img width="1481" height="276" alt="Screenshot 2025-07-30 at 10 41 06" src="https://github.com/user-attachments/assets/e46f6171-7612-401c-b1b9-0376f8672a58" />

garethrees commented:
All looks great, thanks @lucascumsille!
